File: searching/binary_search.py

def binary_search(arr ,e):
    l=0
    u=len(arr)-1
    found=False

    while l<=u:
        mid = int((l + u) / 2)
        if e==arr[mid]:
            globals()['pos']=mid    #position of target elemet is pos=mid  
            return True
        if e>arr[mid]:    #element is in right half
            l=mid+1
        elif e<arr[mid]:  #element is in left half
            u=mid-1       
    return False
result=binary_search([5,7,59,77,87,98,100,799,989,1000],87)

if result:
    print(f"Found at {pos+1}")
else:
    print("Not found")

File: searching/binary_search.c
#include<stdio.h>
#include<conio.h>
int binary_search(int arr [],int size,int num){
    int low,mid,high;
    low=0;
    high=size-1;
    while(low<=high){
        mid=(low+high)/2;
        if(arr[mid]==num) {
            return mid;
        }

        if(num>arr[mid]){   /* req element is  in right sub array*/
            low=mid+1;
        }
        else 
        {
            high=mid-1;
        }
    }
    return -1;
}

int main()
{
    int index,num,size;
    int arr[]={1,2,3,4,5};
    size=sizeof(arr)/sizeof(arr[0]);
    num=4;
    index=binary_search(arr,size,num); 
    printf("Given no %d is found at  position %d ",num,index);

}